app/controllers/cart_controller.rb

```diff
@@ -38,20 +48,25 @@ def remove_item
 
   def update_quantity
     item_id = params[:item_id]
-    # TODO: can only add to quantity if we have enough inventory 
-    # a boolean to indicate whether or not there is enough inventory 
-    # to add one to the cart
-    has_inventory = Item.find(item_id).inventory >= 0 
-    # require 'pry', binding.pry
+    item = Item.find(item_id)
+    item_inventory = item.inventory
+    
+    has_inventory = item_inventory > 0 
     if params[:add] == "true" && has_inventory
       flash[:success] = "You have changed your cart quantity."
       cart.contents[item_id] += 1
+      item.update(inventory: item_inventory - 1)
     elsif params[:add] == "true" && !has_inventory
       flash[:error] = "Not enough in inventory"
-      redirect_to '/cart'
     elsif params[:add] == "false"
       flash[:success] = "You have changed your cart quantity."
-      cart.contents[item_id] -= 1
+      if cart.contents[item_id] == 0
+        flash[:error] = "Cart Empty"
+      else
+        flash[:success] = "You have changed your cart quantity."
+        cart.contents[item_id] -= 1
```

Comment:
Decrease quantity within the cart increases the total inventory for item

Reply:
![](https://media1.giphy.com/media/3o7TKpG9OOUmY7gRy0/giphy.gif?cid=5a38a5a2ec45190ec73b48a0fe12ad47849977a81f82e4c0&rid=giphy.gif)